crates/evm/src/evm/db_commit.rs

```diff
@@ -55,22 +51,35 @@ impl<'a, C: sov_modules_api::Context> DatabaseCommit for EvmDb<'a, C> {
 
             if let Some(ref code) = account_info.code {
                 if !code.is_empty() {
-                    let exists_in_db = self
```

Comment:
even after Fork1 we were checking if code exists in self.code, which means an extra nonexistence proof. which is bad as you can guess in zk :)